### PR TITLE
dehardcodes admin grantable objectives list

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

 dehardcodes admin grantable objectives list by turning it into a var

## Why It's Good For The Game

 i was told by a maintainer 2 do this

## Changelog

:cl:
code: dehardcodes admin grantable objectives list
/:cl:

### CODE REVIEW

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -226,6 +228,7 @@ GLOBAL_LIST(admin_objective_list) //Prefilled admin assignable objective list
 	name = "maroon"
 	var/target_role_type=FALSE
 	martyr_compatible = TRUE
+	admin_grantable = TRUE
```
[the remark this answers is not in the document]
Comment (SmoSmoSmoSmok) [reply code/game/gamemodes/objective.dm]: > i know there's a lot of bad examples here, but when you're adding new vars to a subtype, they generally should go to the bottom, so var/target_role_type here should be at the very bottom, and it's a common theme in the whole file. I won't force you to do it if you don't want to, but if you'd like to help out I would really appreciate it.

done

File: code/game/gamemodes/objective.dm [hunk 2]
```diff
@@ -186,8 +188,9 @@ GLOBAL_LIST(admin_objective_list) //Prefilled admin assignable objective list
 
 /datum/objective/assassinate
 	name = "assasinate"
-	var/target_role_type=FALSE
 	martyr_compatible = TRUE
+	admin_grantable = TRUE
+	var/target_role_type=FALSE
```
Comment (Fikou) [inline code/game/gamemodes/objective.dm:193]:
```suggestion
	var/target_role_type = FALSE
```

File: code/game/gamemodes/objective.dm [hunk 3]
```diff
@@ -205,8 +208,8 @@ GLOBAL_LIST(admin_objective_list) //Prefilled admin assignable objective list
 
 /datum/objective/mutiny
 	name = "mutiny"
-	var/target_role_type=FALSE
 	martyr_compatible = 1
+	var/target_role_type=FALSE
```
Comment (Fikou) [inline code/game/gamemodes/objective.dm:212]:
```suggestion
	var/target_role_type = FALSE
```

File: code/game/gamemodes/objective.dm [hunk 4]
```diff
@@ -224,8 +227,9 @@ GLOBAL_LIST(admin_objective_list) //Prefilled admin assignable objective list
 
 /datum/objective/maroon
 	name = "maroon"
-	var/target_role_type=FALSE
 	martyr_compatible = TRUE
+	admin_grantable = TRUE
+	var/target_role_type=FALSE
```
Comment (Fikou) [inline code/game/gamemodes/objective.dm:232]:
```suggestion
	var/target_role_type = FALSE
```

File: code/game/gamemodes/objective.dm [hunk 5]
```diff
@@ -248,6 +252,7 @@ GLOBAL_LIST(admin_objective_list) //Prefilled admin assignable objective list
 
 /datum/objective/debrain
 	name = "debrain"
+	admin_grantable = TRUE
 	var/target_role_type=0
```
Comment (Fikou) [inline code/game/gamemodes/objective.dm:256]:
```suggestion
	var/target_role_type = FALSE
```

**Review by Fikou — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/objective.dm [hunk 6]
```diff
@@ -934,28 +952,12 @@ GLOBAL_LIST_EMPTY(possible_items_special)
...
-		/datum/objective/custom
-	),/proc/cmp_typepaths_asc)
-
-	for(var/T in allowed_types)
-		var/datum/objective/X = T
-		GLOB.admin_objective_list[initial(X.name)] = T
+	var/list/allowed_types = sort_list(subtypesof(/datum/objective), /proc/cmp_typepaths_asc)
+
+	for(var/datum/objective/goal as anything in allowed_types)
+		if(!initial(goal.admin_grantable))
```
Comment (ShizCalev) [inline code/game/gamemodes/objective.dm:958]: isn't this practically the same as hardcoding them?

Comment (SmoSmoSmoSmok) [reply code/game/gamemodes/objective.dm:958]: > isn't this practically the same as hardcoding them?

how so?

Comment (SmoSmoSmoSmok) [reply code/game/gamemodes/objective.dm:958]: resolved thru discord

**Pull request merged**